src/platform/common/utils/localize.ts

```diff
@@ -18,1186 +26,1360 @@ const packageNlsJsons: Record<string, any> = {
     'zh-tw': require('../../../../package.nls.zh-tw.json')
 };
 
+// TODO: Once vscode-nls supports the web environment, we should:
+// - Remove this function and the conditional afterwards that uses `osType`.
+// - Change the signature of the localize properties in this file to not be functions but constants.
+//   - Instead of `Common.bannerLabelYes()`, make them `Common.bannerLabelYes`.
+export function localizeReplacement(key: string | nls.LocalizeInfo, defaultValue: string): string {
+    if (typeof key !== 'string') {
+        key = key.key;
+    }
+    return getString(key as string, defaultValue);
+}
+const osType = getOSType();
+if (osType === OSType.Unknown) {
+    localize = localizeReplacement;
+}
+
 // External callers of localize use these tables to retrieve localized values.
 
 export namespace Common {
-    export const bannerLabelYes = localize('Common.bannerLabelYes', 'Yes');
-    export const bannerLabelNo = localize('Common.bannerLabelNo', 'No');
-    export const yesPlease = localize('Common.yesPlease', 'Yes, please');
-    export const canceled = localize('Common.canceled', 'Canceled');
-    export const cancel = localize('Common.cancel', 'Cancel');
-    export const ok = localize('Common.ok', 'Ok');
-    export const download = localize('Common.download', 'Download');
-    export const gotIt = localize('Common.gotIt', 'Got it!');
-    export const install = localize('Common.install', 'Install');
-    export const loadingExtension = localize('Common.loadingExtension', 'Jupyter extension loading...');
-    export const openOutputPanel = localize('Common.openOutputPanel', 'Show output');
-    export const noIWillDoItLater = localize('Common.noIWillDoItLater', 'No, I will do it later');
-    export const notNow = localize('Common.notNow', 'Not now');
-    export const doNotShowAgain = localize('Common.doNotShowAgain', 'Do not show again');
-    export const reload = localize('Common.reload', 'Reload');
-    export const moreInfo = localize('Common.moreInfo', 'More Info');
-    export const learnMore = localize('Common.learnMore', 'Learn more');
-    export const and = localize('Common.and', 'and');
-    export const reportThisIssue = localize('Common.reportThisIssue', 'Report this issue');
-    export const clickHereForMoreInfoWithHtml = localize(
-        'Common.clickHereForMoreInfoWithHtml',
-        "Click <a href='{0}'>here</a> for more info."
-    );
+    export const bannerLabelYes = () => localize('Common.bannerLabelYes', 'Yes');
```

Comment:
Changed these to be functions to allow keeping the old behavior in the browser. At first, I had this change within a replacement of the `localize` function itself, but that broke the matching mechanism used by `vscode-nls-dev`. I've got to allow the Node logic to assume `localize` is unchanged, and thus to keep the signature in the other files, these properties now are lambdas.

I will fix this once `vscode-nls` supports the web.